hack/utils.sh

```diff
@@ -0,0 +1,121 @@
+# You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function if_error_exit {
+    ###########################################################################
+    # Description:                                                            #
+    # Validate if previous command failed and show an error msg (if provided) #
+    #                                                                         #
+    # Arguments:                                                              #
+    #   $1 - error message if not provided, it will just exit                 #
+    ###########################################################################
+    if [ "$?" != "0" ]; then
+        if [ -n "$1" ]; then
+            RED="\e[31m"
+            ENDCOLOR="\e[0m"
+            echo -e "[ ${RED}FAILED${ENDCOLOR} ] ${1}"
+        fi
+        exit 1
+    fi
+}
+
+function pass_message {
+    ###########################################################################
+    # Description:                                                            #
+    # show [PASSED] in green and a message as the validation passed.          #
+    #                                                                         #
+    # Arguments:                                                              #
+    #   $1 - message to output                                                #
+    ###########################################################################
+    if [ -z "${1}" ]; then
+        echo "pass_message() requires a message"
+        exit 1
+    fi
+    GREEN="\e[32m"
+    ENDCOLOR="\e[0m"
+    echo -e "[ ${GREEN}PASSED${ENDCOLOR} ] ${1}"
+}
+
+function add_to_path {
+    ###########################################################################
+    # Description:                                                            #
+    # Add directory to path                                                   #
+    #                                                                         #
+    # Arguments:                                                              #
+    #   arg1:  directory                                                      #
+    ###########################################################################
+    [ $# -eq 1 ]
+    if_error_exit "Wrong number of arguments to ${FUNCNAME[0]}"
+
+    local directory="${1}"
+
+    [ -d "${directory}" ]
+    if_error_exit "Directory \"${directory}\" does not exist"
+
+    case ":${PATH:-}:" in
+        *:${directory}:*) ;;
+        *) PATH="${directory}${PATH:+:$PATH}" ;;
+    esac
+}
+
+command_exists() {
```

Comment:
Can you keep the comment here :)

Reply:
sorry it got lost somehow while refactoring, i added it back :)